### PR TITLE
src/common: add debug logs to opentelementry tracing

### PR DESCRIPTION
**Pull request by ideepika**
this is needed to investigate in case there is init, deployment or trace formation problem.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -7376,7 +7376,7 @@ void OSD::ms_fast_dispatch(Message *m)
   if (m->otel_trace.IsValid()) {
     op->osd_parent_span = tracing::osd::tracer.add_span("op-request-created", m->otel_trace);
   } else {
-    op->osd_parent_span = tracing::osd::tracer.start_trace("op-request-created");
+    op->osd_parent_span = tracing::osd::tracer.start_trace("op-request-created", tracing::osd::tracer.is_enabled());
```
Comment (yuvalif) [inline src/osd/OSD.cc:7379]: this is a functional change. what is the reason for that?

Comment (yuvalif) [reply src/osd/OSD.cc:7379]: @ideepika ?

Comment (ideepika) [reply src/osd/OSD.cc:7379]: sorry @yuvalif was traveling, I think this is not needed, I added this when I could not get OSD spans due to improper agent port

Comment (ideepika) [reply src/osd/OSD.cc:7379]: removed this addition.

**Review by yuvalif — APPROVED** [review]

**Comment by ideepika** [comment]
jenkins retest this please

**Comment by ljflores** [comment]
jenkins test make check

**Comment by ljflores** [comment]
Rados suite review: https://pulpito.ceph.com/?branch=wip-yuri3-testing-2023-04-25-1147

Failures, unrelated:
    1. https://tracker.ceph.com/issues/59049
    2. https://tracker.ceph.com/issues/59192
    3. https://tracker.ceph.com/issues/59335
    4. https://tracker.ceph.com/issues/59380
    5. https://tracker.ceph.com/issues/58946
    6. https://tracker.ceph.com/issues/50371
    7. https://tracker.ceph.com/issues/59057
    8. https://tracker.ceph.com/issues/53345

Details:
    1. WaitReplicas::react(const DigestUpdate&): Unexpected DigestUpdate event - Ceph - RADOS
    2. cls/test_cls_sdk.sh: Health check failed: 1 pool(s) do not have an application enabled (POOL_APP_NOT_ENABLED)
    3. Found coredumps on smithi related to sqlite3Found coredumps on smithi related to sqlite3 - Ceph - Cephsqlite
    4. rados/singleton-nomsgr: test failing from "Health check failed: 1 full osd(s) (OSD_FULL)" and "Health check failed: 1 filesystem is offline (MDS_ALL_DOWN)" - Ceph - RADOS
    5. cephadm: KeyError: 'osdspec_affinity' - Ceph - Mgr - Dashboard
    6. Segmentation fault (core dumped) ceph_test_rados_api_watch_notify_pp - Ceph - RADOS
    7. rados/test_envlibrados_for_rocksdb.sh: No rule to make target 'rocksdb_env_librados_test' on centos 8 - Ceph - RADOS
    8. Test failure: test_daemon_restart (tasks.cephadm_cases.test_cli.TestCephadmCLI) - Ceph - Orchestrator

**Comment by ljflores** [comment]
@ideepika @yuvalif looks like this needs a rebase

**Comment by yuriw** [comment]
@ideepika pls rebase and merge at will, it's been tested and approved
ref: https://trello.com/c/YN2r7OyK

**Comment by ljflores** [comment]
jenkins test make check

**Comment by yuriw** [comment]
jenkins test make check

**Comment by ideepika** [comment]
@yuriw I have rebased, waiting for the make check pass. This should not be breaking as it's majorly douting and documentation.

**Comment by yuriw** [comment]
@ideepika stil failed

**Comment by yuriw** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
lua unit tests are crashing. will investigate

File: src/common/tracer.cc [hunk 2]
```diff
@@ -4,12 +4,18 @@
...
 #include "tracer.h"
+#include "common/debug.h"
 
 #ifdef HAVE_JAEGER
 #include "opentelemetry/sdk/trace/batch_span_processor.h"
 #include "opentelemetry/sdk/trace/tracer_provider.h"
 #include "opentelemetry/exporters/jaeger/jaeger_exporter.h"
 
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_osd
```
Comment (yuvalif) [inline src/common/tracer.cc:15]: why osd?

**Comment by yuvalif** [comment]
> lua unit tests are crashing. will investigate

addressed in bbae4a7272e05f9172d01746c435823b122c5d7c

File: src/common/tracer.cc (hunk 2 above)
Comment (yuvalif) [reply src/common/tracer.cc]: fixed here: c1a80016bd984abd66ba5497b2fa7ce3c3251244

**Comment by yuriw** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
@ideepika can you please retest this?

**Comment by ideepika** [comment]
jenkins retest this please

**Comment by yuvalif** [comment]
> > @ideepika can you please retest this?
> 
> sure @yuvalif

i mean. do we need to rerun teuthology against this PR?

**Comment by ideepika** [comment]
> > > @ideepika can you please retest this?
> > 
> > 
> > sure @yuvalif
> 
> i mean. do we need to rerun teuthology against this PR?

oh, I see, sure I can take care of it. Thanks for the commits they look good!

**Comment by ideepika** [comment]
@yuvalif : http://pulpito.front.sepia.ceph.com/ideepika-2023-08-03_12:39:08-rgw-wip-deepika-tracing-testing-distro-default-smithi/

**Comment by ljflores** [comment]
Rados approved. Ref: https://trello.com/c/pMEWaauy/1825-wip-yuri3-testing-2023-08-15-0955

**Pull request merged**